src/pyEQL/solution.py

```diff
@@ -3228,3 +3230,79 @@ def get_salt_list(self):  # pragma: no cover
         :py:meth:`get_salt_dict`
         """
         return self.get_salt_dict()
+
+    @classmethod
+    def from_preset(cls, preset: str) -> Solution:
+        """Instantiate a solution from a preset composition
+
+        Args:
+            preset (str): String representing the desired solution.
+              Valid entries are 'seawater', 'rainwater', 'wastewater',
+              'urine', 'normal saline' and 'Ringers lactate'.
+
+        Returns:
+            A pyEQL Solution object.
+        """
+        # Path to the YAML and JSON files corresponding to the preset
+        yaml_path = os.path.join("presets", f"{preset}.yaml")
+        json_path = os.path.join("presets", f"{preset}.json")
+
+        # Check if the file exists
+        if os.path.exists(yaml_path):
+            preset_path = yaml_path
+        elif os.path.exists(json_path):
+            preset_path = json_path
+        else:
+            logger.error("Invalid solution entered - %s" % preset)
+            raise FileNotFoundError(f"Files '{yaml_path}' and '{json_path} not found!")
+
+        # Create and return a Solution object
+        return cls().from_file(preset_path)
+
+    def to_file(self, file_name: str, extension: str = "yaml") -> None:
+        """Saving to a .yaml or .json file.
+
+        Args:
+            file_name (str): The name of the file with the saved Solution
+            extension (str, optional): The extension of the save file.
+              Valid entries are 'yaml' and 'json'.
+              Defaults to 'yaml'.
```

Comment:
Let's just have a single kwarg -`filename` that is the full path to file, including the extension. You can say in the docstring that the valid extensions are .json or .yaml.

Reply:
Also, in both `to_file` and `from_file`, can you please accept both `Path` and `str` as input?